mllib/src/main/scala/org/apache/spark/mllib/api/python/PythonMLLibAPI.scala

```diff
@@ -763,11 +764,13 @@ private[python] class PythonMLLibAPI extends Serializable {
       maxBins = maxBins,
       categoricalFeaturesInfo = categoricalFeaturesInfo.asScala.toMap)
     val cached = data.rdd.persist(StorageLevel.MEMORY_AND_DISK)
+    // Only done because methods below want an int, not an optional Long
+    val intSeed = getSeedOrDefault(seed).toInt
```

Comment:
This is kind of odd, but I see why you have it.  Maybe it would be a little better to do 
`val intSeed = if (seed == null) Utils.random.nextInt else seed.toInt`

That way it would avoid a slightly biased random Int